### PR TITLE
(mobile): Implement a date picker

### PR DESCRIPTION
**Pull request by HunseopJeong**
[Issue] https://github.com/Samsung/TAU/issues/974
[Problem] Need to implement date picker
[Solution] Implement a date picker

Signed-off-by: Hunseop Jeong <hs85.jeong@samsung.com>

### CODE REVIEW

**Comment by HunseopJeong** [comment]
image

File: src/js/profile/mobile/widget/DatePicker.js [hunk 1]
```diff
@@ -0,0 +1,416 @@
...
+						min: 1,
+						max: 31
+					};
+					spinContainer.classList.add(DatePicker.classes.DAY_CONTAINER);
+				} else if (name === "month") {
+					options = {
+						min: 1,
+						max: MONTH_NAMES.length,
+						labels: MONTH_NAMES.map(function (value) {
+							return value.toUpperCase().substring(0, 3);
```
Comment (DohyungLim) [inline src/js/profile/mobile/widget/DatePicker.js:139]: Why is substring (0,3)?
What is the value value? And I am wondering why .join does it.

**Review by DohyungLim — COMMENTED** [review]
1 comment.

File: src/js/profile/mobile/widget/DatePicker.js (hunk 1 above)
Comment (HunseopJeong) [reply src/js/profile/mobile/widget/DatePicker.js:139]: This function changes the month names.
From MONTH_NAMES = ["January", "February", "March", "April", "May", "June", "July", "August", "September", "October", "November", "December"],
To ["JAN", "FEB", "MAR", ....."DEC].

File: src/css/profile/mobile/common/datepicker.less [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    .font(regular);
+    display: flex;
+    width: 100%;
+    height: 36 * @px_base;
+    font-size: 17 * @sp_base;
+    color: var(--date-picker-header-text-color);
+    line-height: 36 * @px_base;
+    align-items: center;
+    justify-content: center;
+    @media (orientation: landscape) {
```
Comment (singa2000) [inline src/css/profile/mobile/common/datepicker.less]: In the Guide , is it divide according to orientation mode??

File: src/js/profile/mobile/widget/DatePicker.js [hunk 3]
```diff
@@ -0,0 +1,416 @@
...
+				dayInMonth = self._dayInMonth(value.getFullYear(), month);
+
+				if (day > dayInMonth) {
+					day = dayInMonth;
+				}
+
+				value.setDate(day);
+				value.setMonth(month - 1);
+
+				spins.day.option("max", dayInMonth);
```
Comment (singa2000) [inline src/js/profile/mobile/widget/DatePicker.js:309]: Why do you need spins.day.option("max", dayInMonth)?

File: src/js/profile/mobile/widget/DatePicker.js (hunk 1 above)
Comment (DohyungLim) [reply src/js/profile/mobile/widget/DatePicker.js:139]: ok. I understand.

File: src/css/profile/mobile/common/datepicker.less (hunk 2 above)
Comment (HunseopJeong) [reply src/css/profile/mobile/common/datepicker.less]: You are right. I checked another page.

File: src/js/profile/mobile/widget/DatePicker.js (hunk 3 above)
Comment (HunseopJeong) [reply src/js/profile/mobile/widget/DatePicker.js:309]: Need to set the max when changing the month.
Ex) 1. set the 31/Mar
       2. Change the Month to the "Feb"
       3. Should change the "31" to the "28" or "29".

**Review by DohyungLim — APPROVED** [review]
I think looks good.
If you have no other comments, you can merge.

File: src/js/profile/mobile/widget/DatePicker.js [hunk 4]
```diff
@@ -0,0 +1,416 @@
...
+
+			/**
+			* Return Date value or value of one field
+			* @method _getValue
+			* @param {string} type
+			* @return {Date|number}
+			* @member ns.widget.mobile.DatePicker
+			* @protected
+			*/
+			prototype._getValue = function (type) {
```
Comment (TomaszLukawskiSam) [inline src/js/profile/mobile/widget/DatePicker.js]: this method cannot take an argument... Please check the BaseWidget.value method.

Method .value() allows two cases:
- without argument - get value
- with argument - set value

**Review by TomaszLukawskiSam — CHANGES_REQUESTED** [review]
The method `_getValue` needs change

File: src/js/profile/mobile/widget/DatePicker.js (hunk 4 above)
Comment (HunseopJeong) [reply src/js/profile/mobile/widget/DatePicker.js]: `src/js/profile/wearable/widget/wearable/DatePicker.js` also seems to need a fix.
Okay, I will add a new function to separate it.

**Review by TomaszLukawskiSam — APPROVED** [review]
Looks and works good.

**Pull request merged**